packages/block-library/src/block/edit.js

```diff
@@ -6,207 +6,188 @@ import { partial } from 'lodash';
+		 *                                                    user edit reusable
+		 *                                                    blocks.
+		 * @property {boolean}               isFetching       Is the reusable
+		 *                                                    block being fetched.
+		 * @property {boolean}               isSaving         Is the reusable
+		 *                                                    block being saved.
+		 * @property {?boolean}              isTemporary      Is the reusable
+		 *                                                    block temporary.
+		 * @property {Object}                settings         Editor settings.
+		 * @property {?Object}               theReusableBlock The reusable block.
+		 * @property {?string}               title            The reusable block
+		 *                                                    title.
+		 */
+
+		/**
+		 * @param {Function} select
+		 *
+		 * @return {WPReusableBlockEditSelectedData} The selected data.
+		 */
+		( select ) => {
+			const { canUser } = select( 'core' );
+			const { getSettings } = select( 'core/block-editor' );
+			const {
+				__experimentalGetReusableBlock: getReusableBlock,
+				__experimentalIsFetchingReusableBlock: isFetchingReusableBlock,
+				__experimentalIsSavingReusableBlock: isSavingReusableBlock,
+			} = select( 'core/editor' );
+			const reusableBlock = getReusableBlock( ref );
+
+			return {
+				theReusableBlock: reusableBlock,
+				isFetching: isFetchingReusableBlock( ref ),
+				isSaving: isSavingReusableBlock( ref ),
+				isTemporary: reusableBlock?.isTemporary ?? null,
+				blocks: reusableBlock ? parse( reusableBlock.content ) : null,
+				canUpdateBlock:
+					!! reusableBlock &&
+					! reusableBlock.isTemporary &&
+					!! canUser( 'update', 'blocks', ref ),
+				settings: getSettings(),
+				title: reusableBlock?.title ?? null,
 			};
+		},
+		[ ref ]
+	);
+
+	const {
+		__experimentalFetchReusableBlocks: fetchReusableBlocks,
+		__experimentalUpdateReusableBlock: updateReusableBlock,
+		__experimentalSaveReusableBlock: saveReusableBlock,
+	} = useDispatch( 'core/editor' );
+
+	const fetchReusableBlock = partial( fetchReusableBlocks, ref );
+	const onChange = partial( updateReusableBlock, ref );
+	const onSave = partial( saveReusableBlock, ref );
+
+	// Start in edit mode when working with a newly created reusable block.
+	// Start in preview mode when we're working with an existing reusable block.
+	const [ isEditing, setIsEditing ] = useState( isTemporary ?? false );
+
+	// Local state so changes can be made to the reusable block without having to save them.
+	const [ localTitle, setLocalTitle ] = useState( title );
+	const [ localBlocks, setLocalBlocks ] = useState( blocks ?? EMPTY_ARRAY );
+
+	useEffect( () => {
+		if ( ! theReusableBlock ) {
+			fetchReusableBlock();
 		}
-	}
-
-	componentDidMount() {
-		if ( ! this.props.reusableBlock ) {
-			this.props.fetchReusableBlock();
+	}, EMPTY_ARRAY );
```

Comment:
Can you justify this? It seems on the end of being a premature optimization.

```suggestion
	}, [] );
```

Reply:
Good point. I'll avoid the constant for now.